packages/ses/src/compartment-shim.js

```diff
@@ -83,13 +89,143 @@ const assertModuleHooks = compartment => {
+const CompartmentPrototype = {
+  constructor: SharedCompartmentConstructor,
+
+  get globalThis() {
+    return privateFields.get(this).globalObject;
+  },
+
+  /**
+   * @param {string} source is a JavaScript program grammar construction.
+   * @param {{
+   *   transforms: Array<Transform>,
+   *   sloppyGlobalsMode: bool,
+   * }} options.
+   */
+  evaluate(source, options = {}) {
+    // Perform this check first to avoid unecessary sanitizing.
+    if (typeof source !== 'string') {
+      throw new TypeError('first argument of evaluate() must be a string');
+    }
+
+    // Extract options, and shallow-clone transforms.
+    const { transforms = [], sloppyGlobalsMode = false } = options;
+    const localTransforms = [...transforms];
+
+    const {
+      globalTransforms,
+      globalObject,
+      globalLexicals,
+    } = privateFields.get(this);
+    const realmRec = getCurrentRealmRec();
+
+    return performEval(realmRec, source, globalObject, globalLexicals, {
+      globalTransforms,
+      localTransforms,
+      sloppyGlobalsMode,
+    });
+  },
+
+  module(specifier) {
+    if (typeof specifier !== 'string') {
+      throw new TypeError('first argument of module() must be a string');
+    }
+
+    assertModuleHooks(this);
+
+    const { exportsProxy } = getDeferredExports(
+      this,
+      privateFields.get(this),
+      moduleAliases,
+      specifier,
+    );
+
+    return exportsProxy;
+  },
+
+  async import(specifier) {
+    if (typeof specifier !== 'string') {
+      throw new TypeError('first argument of import() must be a string');
+    }
+
+    assertModuleHooks(this);
+
+    return load(privateFields, moduleAnalyses, this, specifier).then(() => {
+      const namespace = this.importNow(specifier);
+      return { namespace };
+    });
+  },
+
+  importNow(specifier) {
+    if (typeof specifier !== 'string') {
+      throw new TypeError('first argument of importNow() must be a string');
+    }
+
+    assertModuleHooks(this);
+
+    const moduleInstance = link(privateFields, moduleAliases, this, specifier);
+    moduleInstance.execute();
+    return moduleInstance.exportsProxy;
+  },
+
+  toString() {
+    return '[object Compartment]';
+  },
+};
+
+/** makeCompartmentConstructor()
+ * Each compartment, including the realm's start compartment, have a unique
+ * compartment constructor that closes over the parent.
+ * Calling the compartment constructor with the full module specifier of a
+ * module static record already loaded by the parent compartment in its module
+ * map allows the child to load that module too, but with a new instance using
+ * only the new compartment's endowments.
+ * Each compartment has a unique `globalThis` and some of its properties are
+ * unique, like the `Function` constructor to ensure that code executed in a
+ * compartment does not escape the compartment.
+ * Lockdown also uses different intrinsics in the start compartment
+ * than in shared compartments.
```

Comment:
Many of the intrinsics of non-start compartments, like `Date` are powerless and shared. But I don't like the adjective `shared` for the non-start compartments themselves, since the compartments themselves are not shared. I don't have an alternate suggestion though.

Reply:
Running with “other”. Considered “non-start”, but too close to the unrelated term “non-starter”.